src/mapping-pmp-v0.ts

```diff
@@ -0,0 +1,228 @@
+import {
+  TokenParamsConfigured,
+  ProjectConfigured
+} from "../generated/IPMPV0/IPMPV0";
+import {
+  PMPProjectConfig,
+  PMPConfig,
+  PMP,
+  Project,
+  Token,
+  PMPLatestState
+} from "../generated/schema";
+import { Address, BigInt, log, store } from "@graphprotocol/graph-ts";
+import { generateContractSpecificId } from "./helpers";
+import { PMP_AUTH_OPTIONS, PMP_PARAM_TYPES } from "./constants";
+
+export function handleTokenParamsConfigured(
+  event: TokenParamsConfigured
+): void {
+  const tokenId = generateContractSpecificId(
+    event.params.coreContract,
+    event.params.tokenId
+  );
+
+  let token = Token.load(tokenId);
+
+  if (!token) {
+    // @dev While PMP is open to any NFT, we only index params on indexed projects
+    return;
+  }
```

Comment:
I think instead of existence check on token, we should do existence check on project.
Technically PMPs can be defined pre-mint, so I think it would be okay create a PMP entity with a token that doesn't yet exist. as far as I can tell, nothing should break in our schema if that is the case...

Reply:
ah! good catch, added the check for project 🙏